schema_cache.rst

```diff
@@ -0,0 +1,78 @@
+.. _schema_cache:
+
+Schema Cache
+============
+
+PostgREST uses the database schema cache to get information for several of its features:
+
+- For :ref:`resource_embedding`, it needs relationships' information, such as related tables, foreign keys, cardinalities, etc.
+- For :ref:`s_procs`, it needs access to their metadata, for example, parameters, return type or volatility. It also needs to verify if the function is overloaded.
+- To do an :ref:`upsert`, it looks for the primary keys.
+- To do an INSERT, it also looks for the primary keys in order to return the Location header.
+- For OPTION requests, it verifies the existence and the data modification capabilities of a table or view.
+- For :ref:`open-api`, it needs information about objects in the database (e.g., object description, columns and parameters).
+
+Retrieving this information directly from the database is costly, that is why PostgREST uses a database schema cache instead. When you change any of the information mentioned above while PostgREST is running, the schema cache turns stale. You then need to reload the schema before you make a request related to these changes, otherwise you'll receive an error instead of the expected response.
+
+For instance, suppose you add a ``cities`` table to your database. This table has a foreign key ``country_id`` referencing an existing ``countries`` table. Then, you make a request to get the data from ``cities`` and their ``countries``:

```

Comment:
For the stale schema cache case, we can add another subheading. That way we add some emphasis and we can link to the section.

I suggest to split the paragraph like this:

```suggestion
Retrieving this information directly from the database is costly, that is why PostgREST uses a database schema cache instead.

The stale schema cache
--------------------------------

When you change any of the information mentioned above while PostgREST is running, the schema cache turns stale. You then need to reload the schema before you make a request related to these changes, otherwise you'll receive an error instead of the expected response.
```

Reply:
Also check the important note here: https://postgrest.org/en/latest/api.html#stored-procedures

The example with the function error should be here as well. That will no longer be true when https://github.com/PostgREST/postgrest/pull/1841 is merged, but for now it can be moved.